prow/cmd/deck/main.go

```diff
@@ -105,10 +108,25 @@ func main() {
 			path: *tideURL,
 		}
 		ta.start()
-		http.Handle("/tide.js", gziphandler.GzipHandler(handleTide(configAgent, ta)))
+		mux.Handle("/tide.js", gziphandler.GzipHandler(handleTide(configAgent, ta)))
 	}
 
-	logger.WithError(http.ListenAndServe(":8080", nil)).Fatal("ListenAndServe returned.")
+	// optionally inject http->https redirect handler
+	if *redirectToHTTPS {
+		redirectMux := http.NewServeMux()
+		redirectMux.Handle("/", func(oldMux *http.ServeMux) http.HandlerFunc {
+			return func(w http.ResponseWriter, r *http.Request) {
+				if r.URL.Scheme != "https" {
+					r.URL.Scheme = "https"
+					http.Redirect(w, r, r.URL.String(), http.StatusMovedPermanently)
+				} else {
+					mux.ServeHTTP(w, r)
```

Comment:
s/mux/oldMux/
This would create an infinite recursion loop as is because this `mux` reference is actually captured so after line 127 `mux==redirectMux` :smile:

Reply:
whoops, yeah originally I wasn't capturing, and then I realized I couldn't swap them, will fix.

Reply:
fixed